extensions-contrib/scan-query/src/main/java/io/druid/query/scan/ScanQueryRunnerFactory.java

```diff
@@ -71,9 +71,7 @@ public ScanQueryRunnerFactory(
           final Query<ScanResultValue> query, final Map<String, Object> responseContext
       )
       {
-        final Number queryTimeout = query.getContextValue(QueryContextKeys.TIMEOUT, null);
-        final long timeoutAt = (queryTimeout == null || queryTimeout.longValue() == 0L)
-                               ? JodaUtils.MAX_INSTANT : System.currentTimeMillis() + queryTimeout.longValue();
+        final long timeoutAt = System.currentTimeMillis() + QueryContexts.getTimeout(query);
```

Comment:
timeoutAt is used only if timeout is specified (https://github.com/jihoonson/druid/blob/8b64ce6ec9e083632bce7bf9fcc365b1d857a41b/extensions-contrib/scan-query/src/main/java/io/druid/query/scan/ScanQueryEngine.java#L161-L182).